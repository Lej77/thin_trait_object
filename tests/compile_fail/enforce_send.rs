use thin_trait_object::*;
use std::rc::Rc;

define_v_table!(
    trait SomeVTable {}
);
impl<T> SomeVTable for Rc<T> {}

fn main() {
    ThinBox::<'_, dyn SomeVTable + Send, _>::new(Rc::new(2), ());
}
