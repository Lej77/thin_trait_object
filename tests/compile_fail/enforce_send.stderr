error[E0277]: `std::rc::Rc<{integer}>` cannot be sent between threads safely
    --> $DIR/enforce_send.rs:10:5
     |
10   |     ThinBox::<'_, dyn SomeVTable + Send, _>::new(Rc::new(2), ());
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::rc::Rc<{integer}>` cannot be sent between threads safely
     |
    ::: $WORKSPACE/src/lib.rs
     |
     |         V::VTable: GetThinTraitVTable<T>,
     |                    --------------------- required by this bound in `thin_trait_object::ThinBox::<'erased_data, V, C>::new`
     |
     = help: the trait `std::marker::Send` is not implemented for `std::rc::Rc<{integer}>`
     = note: required because of the requirements on the impl of `thin_trait_object::auto_traits::EnforceAutoTraits<std::rc::Rc<{integer}>>` for `thin_trait_object::auto_traits::HasSend<()>`
     = note: required because of the requirements on the impl of `thin_trait_object::auto_traits::HasAutoTraits<thin_trait_object::auto_traits::HasSend<()>>` for `std::rc::Rc<{integer}>`
